hoot-core/src/main/cpp/hoot/core/visitors/WorstCircularErrorVisitor.h

```diff
@@ -38,22 +40,36 @@
 namespace hoot
 {
 
-class WorstCircularErrorVisitor : public ElementConstOsmMapVisitor, public SingleStatistic
+class WorstCircularErrorVisitor :
+    public ElementConstOsmMapVisitor,
+    public SingleStatistic,
+    public ElementCriterionConsumer
 {
 public:
   static string className() { return "hoot::WorstCircularErrorVisitor"; }
 
   WorstCircularErrorVisitor(): _worst(-1) {}
+  WorstCircularErrorVisitor(const shared_ptr<ElementCriterion>& filter):
+    _filter(filter),_worst(-1) {}
 
   Meters getWorstCircularError() { return _worst; }
 
   virtual double getStat() const { return _worst; }
 
   virtual void setOsmMap(const OsmMap* map) { _map = map; }
 
+  virtual void addCriterion(const ElementCriterionPtr& e)
```

Comment:
Rather than adding criterions to visitors you could use the `FilteredVisitor` to filter for you. E.g.

```
WorstCircularErrorVisitor worst;
map->visitRo(FilteredVisitor(ArbitraryCriterion(...), worst));
```